### DISCUSSION TITLE
could you add support for ligatures?

### DISCUSSION THREAD

**Discussion by ghost** [git-for-windows/git#3025] [Q&A]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

system info is irrelevant, as this is not a bug report, but rather an issue with support for certain font features; fira code retina has a bunch of different valid uses of ligatures in a monospace font
![https://github.com/tonsky/FiraCode/raw/master/extras/ligatures.png shows the difference between ligatures off and ligatures on](https://github.com/tonsky/FiraCode/raw/master/extras/ligatures.png)

**Comment by rimrul** [comment]
Where exactly do you want ligatures?

**Comment by ghost** [comment]
in git bash, and git cmd if possible

**Comment by dscho** [comment]
Git Bash is (usually) backed up by [MinTTY](https://github.com/mintty/mintty/), Git CMD by the native CMD window. Please do feel free to report your feature request with those projects/products.

**Comment by rimrul** [comment]
There should already be ligature support in MinTTY to some capacity (see mintty/mintty#605).

**Comment by rimrul** [comment]
Modern Conhost on Windows 10 schould also offer [some ligature support](https://github.com/microsoft/terminal/issues?q=is%3Aissue+ligature) as well.

**Comment by kabel** [comment]
Adding `-o LigaturesSupport=2` to the MinTTY invocation will enable ligatures. Example:

```
C:\Program Files\Git\usr\bin\mintty.exe --nodaemon -o AppID=GitForWindows.Bash -o AppLaunchCmd="C:\Program Files\Git\git-bash.exe" -o AppName="Git Bash" -o LigaturesSupport=2 -i "C:\Program Files\Git\git-bash.exe" --store-taskbar-properties -- /usr/bin/bash --login -i
```
![image](https://user-images.githubusercontent.com/675956/106943882-951d3880-66eb-11eb-87d4-24438b602596.png)

**Comment by rimrul** [comment]
>Adding `-o LigaturesSupport=2` to the MinTTY invocation will enable ligatures.

The 2 had me a little confused here, because it sounds like it should be a boolean option. And lo and behold, there's a little more to it:

> **Interactive Ligatures support (LigaturesSupport=0):**
> By default, ligatures, as supported by the selected font, are rendered if they are output to the terminal in one chunk. When this option is set =1, mintty redisplays the left part of the line whenever a character is output, so ligatures are also supported while being input.With LigaturesSupport=2, mintty also redisplays the previous cursor line after the cursor is moved.

So this option only changes the way MinTTY behaves when the input could be ligatures.
There's also _Ligatures=2_ to enable additional Ligatures:

> **Ligatures supported (Ligatures=1):**
> This setting can affect the set of ligatures applied, as supported by the selected font.When this option is set =1, the default set of ligatures is applied.With a value greater than 1, additional ligatures are enabled, e.g. turning "<-", "->", "<--", "-->" into arrows, and "<=", ">=" into less/greater or equal symbols. Note the ambiguity of ligature transformation as e.g. "<=" could as well be meant to be an arrow. There is currently no mechanism to affect ligature transformation in more detail.

And the way to enable/disable ligature support is through the _FontRender_ option.

> **Ligatures support**
> These options affect support for ligatures.They are not capable of disabling ligatures, however, as those are applied by Windows font handling. Setting _FontRender=textout_ disables Uniscribe, including ligatures support.
